mipengine/controller/api/algorithms_endpoint.py

```diff
@@ -50,45 +50,38 @@ async def get_algorithms() -> str:
 
 @algorithms.route("/algorithms/<algorithm_name>", methods=["POST"])
 async def post_algorithm(algorithm_name: str) -> str:
-    logging.info(f"Algorithm execution with name {algorithm_name}.")
 
-    request_body = await request.data
+    # Parse the request body to AlgorithmRequestDTO
+    algorithm_request_dto = None
+    try:
```

Comment:
The way the api is implemented right now we need to have a bare except everywhere. If something unexpected happens, we need to throw UnexpectedException that will be translated to a 500 error for quart.

Maybe we can wrap the whole post_algorithm code with a try and (generic) except for any cases that we haven't considered.

Reply:
concerning post_algorithm function, there is now a try/catchall around every line of code.